coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -118,9 +120,14 @@ private static String constructSparkSQLWithExplicitAlias(RelNode sparkRelNode, L
     // Create temporary objects r and rewritten to make debugging easier
     SqlImplementor.Result r = rel2sql.visitChild(0, sparkRelNode);
     SqlNode rewritten = r.asStatement().accept(new SparkSqlRewriter());
-    // Use a second pass visit to add explicit alias names
-    SqlNode aliasAdded = rewritten.accept(new AddExplicitAlias(aliases));
-    return aliasAdded.toSqlString(SparkSqlDialect.INSTANCE).getSql();
+    SqlNode ret = rewritten;
+    // Use a second pass visit to add explicit alias names,
+    // only do this when it's not a select star case,
+    // since for select star we don't need to add any explicit aliases
+    if (rewritten.getKind() == SqlKind.SELECT && ((SqlSelect) rewritten).getSelectList() != null) {
+      ret = rewritten.accept(new AddExplicitAlias(aliases));
+    }
+    return ret.toSqlString(SparkSqlDialect.INSTANCE).getSql();
```

Comment:
Can we just use `rewritten` for simplicity? like:
```
    SqlNode rewritten = r.asStatement().accept(new SparkSqlRewriter());
    // Use a second pass visit to add explicit alias names,
    // only do this when it's not a select star case,
    // since for select star we don't need to add any explicit aliases
    if (rewritten.getKind() == SqlKind.SELECT && ((SqlSelect) rewritten).getSelectList() != null) {
      rewritten.accept(new AddExplicitAlias(aliases));
    }
    return rewritten.toSqlString(SparkSqlDialect.INSTANCE).getSql();
```

Reply:
Sure, refactored.